addons/website/static/src/client_actions/website_preview/website_preview.js

```diff
@@ -479,6 +482,26 @@ export class WebsitePreview extends Component {
         // it is set to undefined again.
         this.websiteService.websiteRootInstance = undefined;
     }
+    /**
+     * Handles refreshing while the website preview is active.
+     * Makes it possible to stay in the backend after an F5 or CTRL-R keypress.
+     *
+     * @param  {KeyboardEvent} ev
+     * @private
+     */
+    _onKeydownRefresh(ev) {
+        const ctrlR = (ev.metaKey || ev.ctrlKey) && ev.key === "r";
+        const f5 = ev.key === "F5";
+        if (!ctrlR && !f5) {
+            return;
+        }
+        ev.preventDefault();
+        const path = this.websiteService.contentWindow.location;
+        const debugMode = this.env.debug ? `?debug=${odoo.debug}` : "";
```

Comment:
Note that in 16.4 since https://github.com/odoo/odoo/commit/59f49d6a8c856a5d8ca28dcfd7840baaa1c7672f I think this won't do anything:
- If you have the debug in the URL, it will reload with it in the URL, right?
- If you don't have the debug in the URL, but you are in debug (server side) it will return false 

I created a task related to this commit for the website preview, see https://www.odoo.com/web#id=3475574&menu_id=4720&cids=1&action=333&active_id=974&model=project.task&view_type=form

I'll let you decide and check what's the best to do with this forward port later

Well, I just saw that you considered that in your commit message actually, but I don't think it's needed here and in 16.4, no?

Reply:
> If you have the debug in the URL, it will reload with it in the URL, right?

I think we talked about it at the office but just to be sure:

Let's say you go to URL `/@/?debug=asset` what this will do is pass the param in the URL of the iframe, but it will not keep it in the backend. Which means in 16.4, you will loose the JS debug mode, which makes tracebacks less explicit and removes some technical functionality (like the "Show optimised images" toggle in the media dialog).

Another issue is that even if you do "/web?debug=assets#..." to get to the client action, you will loose the debug mode from the URL as the iframe loads without it, then replaces the URL of the top window with the one from the iframe, without the param.

But when the iframe is first loaded, the odoo.debug variable still holds the previous debug mode, so we can restore the param in the URL.

It's not needed in 16.0-16.3 but I thought I'd leave it there in order to keep the commit the same in each version 😄